narwhals/_dask/dataframe.py

```diff
@@ -258,133 +258,160 @@ def sort(
             )
-            extra = []
-            for left_key, right_key in zip(left_on, right_on):  # type: ignore[arg-type]
-                if right_key != left_key and right_key not in self.columns:
-                    extra.append(right_key)
-                elif right_key != left_key:
-                    extra.append(f"{right_key}_right")
-            return self._with_native(result_native.drop(columns=extra))
+            .drop(columns=key_token)
+        )
 
-        if how == "full":
-            # dask does not retain keys post-join
-            # we must append the suffix to each key before-hand
-
-            # help mypy
-            assert left_on is not None  # noqa: S101
-            assert right_on is not None  # noqa: S101
-
-            right_on_mapper = _remap_full_join_keys(left_on, right_on, suffix)
-            other_native = other.native.rename(columns=right_on_mapper)
-            check_column_names_are_unique(other_native.columns)
-            right_on = list(right_on_mapper.values())  # we now have the suffixed keys
-            return self._with_native(
-                self.native.merge(
-                    other_native,
-                    left_on=left_on,
-                    right_on=right_on,
-                    how="outer",
-                    suffixes=("", suffix),
-                )
+    def _join_semi(
+        self, other: Self, *, left_on: Sequence[str], right_on: Sequence[str]
+    ) -> Self:
+        other_native = (
+            select_columns_by_name(
+                other.native,
+                list(right_on),
+                self._backend_version,
+                self._implementation,
             )
-
+            .rename(  # rename to avoid creating extra columns in join
+                columns=dict(zip(right_on, left_on))
+            )
+            .drop_duplicates()  # avoids potential rows duplication from inner join
+        )
         return self._with_native(
             self.native.merge(
+                other_native, how="inner", left_on=left_on, right_on=left_on
+            )
+        )
+
+    def _join_anti(
+        self, other: Self, *, left_on: Sequence[str], right_on: Sequence[str]
+    ) -> Self:
+        indicator_token = generate_temporary_column_name(
+            n_bytes=8, columns=[*self.columns, *other.columns]
+        )
+
+        other_native = (
+            select_columns_by_name(
                 other.native,
-                left_on=left_on,
-                right_on=right_on,
-                how=how,
-                suffixes=("", suffix),
+                column_names=list(right_on),
+                backend_version=self._backend_version,
+                implementation=self._implementation,
             )
+            # rename to avoid creating extra columns in join
+            .rename(columns=dict(zip(right_on, left_on)))
+            .drop_duplicates()
+        )
+        df = self.native.merge(
+            other_native,
+            how="outer",
```

Comment:
Spotted that we are wasted some cycles - this can be a left join (with the same filter below). Fixing that